tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc

```diff
@@ -415,7 +415,10 @@ std::vector<uint8> NVPTXCompiler::CompileGpuAsmOrGetCachedResult(
                   "using $PATH.",
                   hlo_module_config);
             }
-          } else {
+          } else if (maybe_cubin.status().code() !=
```

Comment:
XLA used to silently fallback to the driver when ptxas couldn't be found or compilation failed. The silent fallback behavior led to several bugs reported by users that were hard to reproduce and diagnose. We hence decided to turn ptxas issues into fatal errors (few people notice warnings in logs) and allow manual overwrite by the flag (--xla_gpu_unsafe_fallback_to_driver_on_ptxas_not_found).
Have you considered to introduce a flag for the _ptxas_too_old case? I think that would be the better option.

Reply:
Here it isn't a real compilation failure. Just that PTXAS doesn't know a specific SM version.

Where can I have more information about the bug that you had that triggered this decision?

I'll think about your suggestion and come back about it.

Reply:
Quick follow up:
For currently supported GPU, there is no change. The new fallback is only for new GPUs. So when an (current or old) container is used on a newer GPU, the fallback is used.